src/main/java/net/sf/jabref/logic/groups/KeywordGroup.java

```diff
@@ -268,41 +267,38 @@ public static boolean containsWord(String word, String text) {
      * possible if the search expression is not a regExp.
      */
     private void removeMatches(BibEntry entry) {
-        if (!entry.hasField(searchField)) {
-            return; // nothing to modify
-        }
-        String content = entry.getField(searchField);
-        StringBuffer sbOrig = new StringBuffer(content);
-        StringBuffer sbLower = new StringBuffer(content.toLowerCase());
-        StringBuffer haystack = caseSensitive ? sbOrig : sbLower;
-        String needle = caseSensitive ? searchExpression
-                : searchExpression.toLowerCase();
-        int i;
-        int j;
-        int k;
-        final String separator = jabRefPreferences.get(JabRefPreferences.KEYWORD_SEPARATOR);
-        while ((i = haystack.indexOf(needle)) >= 0) {
-            sbOrig.replace(i, i + needle.length(), "");
-            sbLower.replace(i, i + needle.length(), "");
-            // reduce spaces at i to 1
-            j = i;
-            k = i;
-            while (((j - 1) >= 0) && (separator.indexOf(haystack.charAt(j - 1)) >= 0)) {
-                --j;
-            }
-            while ((k < haystack.length()) && (separator.indexOf(haystack.charAt(k)) >= 0)) {
-                ++k;
+        entry.getFieldOptional(searchField).ifPresent(content -> {
+            StringBuffer sbOrig = new StringBuffer(content);
+            StringBuffer sbLower = new StringBuffer(content.toLowerCase());
+            StringBuffer haystack = caseSensitive ? sbOrig : sbLower;
+            String needle = caseSensitive ? searchExpression : searchExpression.toLowerCase();
```

Comment:
What the hell is that for a method?
![needlehaystack](https://cloud.githubusercontent.com/assets/320228/17169348/4cd11ee6-53e7-11e6-8746-e324f255a3e8.jpg)